### ISSUE TITLE
feat(patch): Spotify Premium like features

### ISSUE THREAD

**Issue by GioPan04** [revanced/revanced-patches#1015]
### Application

Spotify (`com.spotify.music`)

### Issue

See patch section

### Patch

- Play any songs, currently you can search songs but only play playlists and you can't choose a song in them.
- Songs in playlist can only be played in shuffle
- Remove audio ads
- Unlimited songs skips

### Motivation

Spotify is a big app, used by many. Currently there aren't any good alternatives for android. There are some patched version online but they are all closed source as much as I know, having an open source ones could be a great thing.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by cyberboh** [comment]
`Download music just for premium` and it `server-sided`. But at leats im gratefully if `ads` also disabled

**Comment by vasile-artgames** [comment]
You could try xManager to download the modded spotify app, it has all the features you need https://github.com/xManager-v2/xManager-Spotify

**Comment by SodaWithoutSparkles** [comment]
Idea: would it be possible to invoke external downloader? like revanced yt does? it seems that yt-dlp supports spotify, and other projects lime [spotDL](https://github.com/spotDL/spotify-downloader) exist

**Comment by GioPan04** [comment]
@SodaWithoutSparkles yeah you could call an external downloader, but offline play with all the metadata would be more difficult.

**Comment by shidozimenoshi** [comment]
Hey, i have an idea. maybe it works. Have a look at this open source project- https://github.com/Shabinder/SpotiFlyer
This app downloads music in mp3 format from spotify. Maybe you guys could find something in its code.
